// zenn-cli/dist/cli/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var arg_1 = __importDefault(require("arg"));
var safe_1 = __importDefault(require("colors/safe"));
var update_notifier_1 = __importDefault(require("update-notifier"));
var pkg = require("../../package.json"); // refer package.json from dist/cli/index.js
var commands = {
    preview: function () { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, Promise.resolve().then(function () { return __importStar(require("./preview")); }).then(function (i) { return i.exec; })];
            case 1: return [2 /*return*/, _a.sent()];
        }
    }); }); },
    help: function () { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, Promise.resolve().then(function () { return __importStar(require("./help")); }).then(function (i) { return i.exec; })];
            case 1: return [2 /*return*/, _a.sent()];
        }
    }); }); },
    init: function () { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, Promise.resolve().then(function () { return __importStar(require("./init")); }).then(function (i) { return i.exec; })];
            case 1: return [2 /*return*/, _a.sent()];
        }
    }); }); },
    "new:article": function () { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, Promise.resolve().then(function () { return __importStar(require("./new-article")); }).then(function (i) { return i.exec; })];
            case 1: return [2 /*return*/, _a.sent()];
        }
    }); }); },
    "new:book": function () { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, Promise.resolve().then(function () { return __importStar(require("./new-book")); }).then(function (i) { return i.exec; })];
            case 1: return [2 /*return*/, _a.sent()];
        }
    }); }); },
};
var args = arg_1.default({
    "--version": Boolean,
    // alias
    "-v": "--version",
}, {
    permissive: true,
});
var command = args._[0] || "preview";
// notify package update
var notifier = update_notifier_1.default({
    pkg: pkg,
    updateCheckInterval: 1000 * 60 * 60 * 24,
});
notifier.notify({
    message: "CLIã«å¤‰æ›´ãŒã‚ã‚Šã¾ã™ {currentVersion} â†’ {latestVersion} ðŸ’¡\n`npm install zenn-cli@latest`ã§æ›´æ–°ã—ã¦ãã ã•ã„",
});
if (args["--version"]) {
    console.log("zenn-cli v" + pkg.version);
    process.exit(0);
}
if (!commands[command]) {
    console.log(safe_1.default.red('ðŸ˜¿è©²å½“ã™ã‚‹CLIã‚³ãƒžãƒ³ãƒ‰ãŒå­˜åœ¨ã—ã¾ã›ã‚“ã€‚ã‚³ãƒžãƒ³ãƒ‰ä¸€è¦§ã¯"zenn help"ã§ç¢ºèªã§ãã¾ã™'));
    process.exit(1);
}
commands[command]().then(function (exec) { return exec(args._.slice(1)); });
