// zenn-markdown-html/lib/utils/option-md-custom-block.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.optionCustomBlock = void 0;
var md_base_1 = require("./md-base");
// ref: https://github.com/posva/markdown-it-custom-block
// e.g.
// @[youtube](youtube-video-id)
exports.optionCustomBlock = {
    youtube: function (videoId) {
        if (!(videoId === null || videoId === void 0 ? void 0 : videoId.match(/^[a-zA-Z0-9_-]+$/))) {
            return "YouTubeのvideoIDが不正です";
        }
        return "<div class=\"embed-youtube\"><iframe src=\"https://www.youtube.com/embed/" + md_base_1.escapeHtml(videoId) + "\" allowfullscreen loading=\"lazy\"></iframe></div>";
    },
    slideshare: function (key) {
        if (!(key === null || key === void 0 ? void 0 : key.match(/^[a-zA-Z0-9_-]+$/))) {
            return "Slide Shareのkeyが不正です";
        }
        return "<div class=\"embed-slideshare\"><iframe src=\"https://www.slideshare.net/slideshow/embed_code/key/" + md_base_1.escapeHtml(key) + "\" scrolling=\"no\" allowfullscreen loading=\"lazy\"></iframe></div>";
    },
    speakerdeck: function (key) {
        if (!(key === null || key === void 0 ? void 0 : key.match(/^[a-zA-Z0-9_-]+$/))) {
            return "Speaker Deckのkeyが不正です";
        }
        return "<div class=\"embed-speakerdeck\"><iframe src=\"https://speakerdeck.com/player/" + md_base_1.escapeHtml(key) + "\" scrolling=\"no\" allowfullscreen allow=\"encrypted-media\" loading=\"lazy\"></iframe></div>";
    },
    jsfiddle: function (str) {
        if (!(str === null || str === void 0 ? void 0 : str.match(/^(http|https):\/\/jsfiddle\.net\/[a-zA-Z0-9_,\/\-]+$/))) {
            return "jsfiddleのURLが不正です";
        }
        // URLを~/embedded/とする
        // ※ すでにembeddedもしくはembedが含まれるURLが入力されている場合は、そのままURLを使用する。
        var url = str;
        if (!url.includes("embed")) {
            url = url.endsWith("/") ? url + "embedded/" : url + "/embedded/";
        }
        return "<div class=\"embed-jsfiddle\"><iframe src=\"" + url + "\" scrolling=\"no\" frameborder=\"no\" allowfullscreen allowtransparency=\"true\" loading=\"lazy\"></iframe></div>";
    },
    codepen: function (str) {
        if (!(str === null || str === void 0 ? void 0 : str.match(/^https:\/\/codepen\.io\/[a-zA-Z0-9]/))) {
            return "CodePenのURLが不正です";
        }
        var url = new URL(str.replace("/pen/", "/embed/"));
        url.searchParams.set("embed-version", "2");
        return "<div class=\"embed-codepen\"><iframe src=\"" + url + "\" scrolling=\"no\" scrolling=\"no\" frameborder=\"no\" allowtransparency=\"true\" loading=\"lazy\"></iframe></div>";
    },
    // クライアントでhttps://platform.twitter.com/widgets.jsを読み込む必要あり
    tweet: function (str) {
        if (!(str === null || str === void 0 ? void 0 : str.match(/^https:\/\/twitter.com\/[a-zA-Z0-9\_\-\/]+$/))) {
            return "ツイートページのURLを指定してください";
        }
        return "<div class=\"embed-tweet tweet-container\"><blockquote class=\"twitter-tweet\"><a href=\"" + str + "\"></a></blockquote></div>";
    },
};
